components/feature/prompts/src/main/java/mozilla/components/feature/prompts/share/SharePrompt.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.share
+
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.ACTION_SEND
+import android.content.Intent.EXTRA_SUBJECT
+import android.content.Intent.EXTRA_TEXT
+import android.content.Intent.EXTRA_TITLE
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.concept.engine.prompt.PromptRequest.Share
+import mozilla.components.concept.engine.prompt.ShareData
+import mozilla.components.feature.prompts.Prompter
+import mozilla.components.support.ktx.R
+
+internal class SharePrompt(
+    private val context: Context,
+    private val feature: Prompter
+) {
+
+    fun handleShareRequest(promptRequest: Share, session: SessionState) {
+        val sendIntent = promptRequest.data.asSendIntent()
+        try {
+            displayShareSheet(sendIntent, session)
+        } catch (e: ActivityNotFoundException) {
+            feature.onCancel(session.id)
+        }
+    }
+
+    private fun displayShareSheet(sendIntent: Intent, session: SessionState) {
```

Comment:
We have `Context.share` (and it seems like you saw it). Does it not work for this case, and/or modify it to work?